distributed-job-manager/src/main/kotlin/ru/fix/distributed/job/manager/ManagerState.kt

```diff
@@ -0,0 +1,87 @@
+package ru.fix.distributed.job.manager
+
+import org.apache.logging.log4j.kotlin.Logging
+import ru.fix.stdlib.concurrency.events.ReducingEventAccumulator
+import java.util.concurrent.atomic.AtomicReference
+
+/**
+ * Gives information about Manager state (leadership, shutdown status, necessity for rebalance)
+ * */
+class ManagerState {
+
+    companion object : Logging
+
+    private val currentState = AtomicReference(State.IS_NOT_LEADER)
+
+    private val newRebalanceNeededAccumulator = ReducingEventAccumulator<ManagerEvent, Boolean> { _, newEvent ->
```

Comment:
Seems like extracting ManagerState with internal AtomicReference currentState from Manager is overhead that complicate readability. 
Suggestion: do not extract internal state from Manager to separate class. 

Manager class responsibility will be:
1. becoming single leader in the cluster
2. activate re-balance if needed. 

Cleaner class stays almost the same, it will depends only on  Manager##isActiveLeader()

Rebalancer class simply provides Rebalancer::reassignAndBalanceTasks() method

Manager: 
Uses leaderLatch, aliveWorkersCache, eventAccumulator?  cleaner and rebalancer.
Also manager maintain thread that checks eventAccumulator and calls Rebalancer:reassignAndBalanceTasks.


This way:
- Only manager knows about the need to compress zk events  into single rebalance
- Only manager responsible for single thread where to invoke rebalance via eventReducer
- Manager binds zk events to actions like rebalance. 
- Rebalancer does not overwhelmed  with compressing zk events ideas, and only responsible for rebalances.

Reply:
ManagerState logic moved into Manager class